src/assets/dpos.ts

```diff
@@ -0,0 +1,190 @@
+
+	for (const account of accounts) {
+		if (account.dpos.delegate.username !== '') {
+			const validatorAddress = getLisk32AddressFromAddress(account.address);
+			const validator: ValidatorEntry = {
+				address: validatorAddress,
+				name: account.dpos.delegate.username,
+				blsKey: INVALID_BLS_KEY,
+				proofOfPossession: DUMMY_PROOF_OF_POSSESSION,
+				generatorKey: '',
+				lastGeneratedHeight: 0,
+				isBanned: true,
+				pomHeights: [],
+				consecutiveMissedBlocks: 0,
+				lastCommissionIncreaseHeight: snapshotHeight,
+				commission: 10000,
+				sharingCoefficients: {
+					tokenID: TOKEN_ID_LSK,
+					coefficient: Q96_ZERO,
+				},
+			};
+			if (validatorKeys[validatorAddress]) {
+				validator.generatorKey = validatorKeys[validatorAddress];
+			} else {
+				validator.generatorKey = INVALID_ED25519_KEY;
+			}
+			validator.lastGeneratedHeight = account.dpos.delegate.lastForgedHeight;
+			validator.pomHeights = account.dpos.delegate.pomHeights;
+			validator.consecutiveMissedBlocks = account.dpos.delegate.consecutiveMissedBlocks;
+			validators.push(validator);
+		}
+	}
+	return validators;
+};
+
+export const getSentVotes = async (account: Account): Promise<SentVote[]> => {
+	const sentVotes = account.dpos.sentVotes.map(vote => ({
+		...vote,
+		delegateAddress: getLisk32AddressFromAddress(vote.delegateAddress),
+		voteSharingCoefficients: [
+			{
+				tokenID: TOKEN_ID_LSK,
+				coefficient: Q96_ZERO,
+			},
+		],
+	}));
+
+	return sentVotes;
+};
+
+export const createVotersArray = async (accounts: Account[]): Promise<Voter[]> => {
+	const voters: Voter[] = [];
+	for (const account of accounts) {
+		if (account.dpos.sentVotes.length || account.dpos.unlocking.length) {
+			const voter: Voter = {
+				address: getLisk32AddressFromAddress(account.address),
+				sentVotes: await getSentVotes(account),
+				pendingUnlocks: account.dpos.unlocking.map(unlock => ({
+					delegateAddress: getLisk32AddressFromAddress(unlock.delegateAddress),
+					amount: unlock.amount,
+					unvoteHeight: unlock.unvoteHeight,
+				})),
+			};
+			voters.push(voter);
+		}
+	}
+	return voters;
+};
+
+export const createGenesisDataObj = async (
+	accounts: Account[],
+	delegatesVoteWeights: DecodedVoteWeights,
+	snapshotHeight: number,
+): Promise<GenesisDataEntry> => {
+	const r = Math.ceil((snapshotHeight - HEIGHT_PREVIOUS_SNAPSHOT_BLOCK) / ROUND_LENGTH);
+	const topDelegates: DelegateWeight[] | undefined = delegatesVoteWeights.voteWeights
+		.find((voteWeight: VoteWeight) => voteWeight.round === r - 2)
+		?.delegates.slice(0, 101);
+
+	const initDelegates: Buffer[] = [];
+	if (topDelegates?.length) {
+		const accountMap = new Map(accounts.map(account => [account.address.toString('hex'), account]));
+
+		topDelegates.forEach((delegate: DelegateWeight) => {
+			const account = accountMap.get(delegate.address.toString('hex'));
+			if (account && !account.dpos.delegate.isBanned) {
+				initDelegates.push(delegate.address);
+			}
+		});
```

Comment:
Since the `isBanned` flag is verified here, the `initDelegates` list can possibly result shorter than the intended count of `101`. Please adjust the implementation accordingly.